### PR TITLE
Cache

### PR DESCRIPTION
**Pull request by jbants**
Updates caching to use python3 LRUcache
Fixes tests
Formatting and code cleanup

### CODE REVIEW

File: stac_validator/stac_utilities.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def item_schema_url(cls, version, filename="stac-item.json"):
         return cls(version, "json-schema", filename).ITEM_URL
 
     @classmethod
-    def item_geojson_schema_url(cls, version, filename="geojson.json"):
+    def geojson_schema_url(cls, version, filename="geojson.json"):
```
Comment (gridcell) [inline stac_validator/stac_utilities.py]: `geojson_schema_url` name looks more to be a property than a method

File: stac_validator/stac_validator.py [hunk 2]
```diff
@@ -75,86 +74,59 @@ def __init__(self, stac_file, version="master", loglevel="CRITICAL"):
...
-            logging.debug("Copying STAC catalog spec from local file to cache")
-            stac_catalog_schema = json.dumps(stac_catalog)
-            fp.write(stac_catalog_schema)
-            cache[catalog_key] = stac_catalog_schema
+        # Write the stac file to a filepath. used as absolute links for geojson schmea
+        if spec_name == "geojson":
+            file_name = os.path.join(self.dirpath, "geojson.json")
+        else:
+            file_name = os.path.join(
+                self.dirpath, f"{spec_name}_{self.stac_version.replace('.','_')}.json"
```
Comment (gridcell) [inline stac_validator/stac_validator.py]: `...stac_version.replace('.','_').....` <--- space after period.

File: stac_validator/stac_validator.py [hunk 3]
```diff
@@ -75,86 +74,59 @@ def __init__(self, stac_file, version="master", loglevel="CRITICAL"):
...
-            stac_item_schema = json.dumps(stac_item)
-            fp.write(stac_item_schema)
-            cache[item_key] = stac_item_schema
-
-        with open(os.path.join(self.dirpath, "stac-catalog.json"), "w") as fp:
-            logging.debug("Copying STAC catalog spec from local file to cache")
-            stac_catalog_schema = json.dumps(stac_catalog)
-            fp.write(stac_catalog_schema)
-            cache[catalog_key] = stac_catalog_schema
+        # Write the stac file to a filepath. used as absolute links for geojson schmea
```
Comment (gridcell) [inline stac_validator/stac_validator.py]: spelling `schmea`

File: stac_validator/stac_validator.py [hunk 4]
```diff
@@ -75,86 +74,59 @@ def __init__(self, stac_file, version="master", loglevel="CRITICAL"):
...
-        stac_schema = json.loads(schema)
         try:
             if "title" in stac_schema and "item" in stac_schema["title"].lower():
                 logger.debug("Changing GeoJson definition to reference local file")
                 # rewrite relative reference to use local geojson file
                 stac_schema["definitions"]["core"]["allOf"][0]["oneOf"][0]["$ref"] = (
-                    "file://"
-                    + cache["geojson_resolver"]
-                    + "/geojson.json#definitions/feature"
+                    "file://" + self.dirpath + "/geojson.json#definitions/feature"
```
Comment (gridcell) [inline stac_validator/stac_validator.py]: string formatting

`f"file://{self.dirpath}/geojson.json#definitions/feature"`

File: VERSION [hunk 5]
```diff
@@ -1 +1 @@
-0.0.1
\ No newline at end of file
+0.0.2
```
Comment (brianbancroft) [inline VERSION]: I might be missing something, but this doesn't appear to be in-line with https://github.com/sparkgeo/stac-validator/releases

File: stac_validator/stac_utilities.py (hunk 1 above)
Comment (brianbancroft) [reply stac_validator/stac_utilities.py]: prepend with `get_`?

File: stac_validator/stac_validator.py [hunk 6]
```diff
@@ -75,86 +74,59 @@ def __init__(self, stac_file, version="master", loglevel="CRITICAL"):
...
-                stac_item = requests.get(StacVersion.item_schema_url(version)).json()
-                stac_catalog = requests.get(
-                    StacVersion.catalog_schema_url(version)
-                ).json()
-            except Exception as error:
-                logger.exception("STAC Download Error")
-                raise VersionException(
-                    f"Could not download STAC specification files for version: {version}"
-                )
+            spec_name = "item"
```
Comment (brianbancroft) [inline stac_validator/stac_validator.py]: This might be a good place to use a case statement, instead?

File: stac_validator/stac_validator.py [hunk 7]
```diff
@@ -180,7 +153,15 @@ def validate_json(self, stac_content, schema):
             logger.exception("STAC error")
             return False, f"{error}"
 
-    def _update_status(self, old_status, new_status):
+    @staticmethod
+    def _update_status(old_status, new_status):
```
Comment (brianbancroft) [inline stac_validator/stac_validator.py]: A question: This is not the first place I've seen a method name prepended with an underscore. Can you speak to the convention and what it denotes?

File: VERSION (hunk 5 above)
Comment (jbants) [reply VERSION]: Ahh. Good catch

File: stac_validator/stac_validator.py (hunk 7 above)
Comment (jbants) [reply stac_validator/stac_validator.py]: As I understand it, its simply meant to show the method should be considered private (even though it is actually public).

File: stac_validator/stac_validator.py (hunk 6 above)
Comment (jbants) [reply stac_validator/stac_validator.py]: python doesn't have case statements. I supposed we could do something with dictionaries for when we add extensions.

**Pull request merged**